flutter/lib/src/event_processor/platform_exception_event_processor.dart

```diff
@@ -0,0 +1,115 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+import 'package:sentry/sentry.dart';
+import 'package:stack_trace_parser/stack_trace_parser.dart';
+
+class AndroidPlatformExceptionEventProcessor implements EventProcessor {
+  const AndroidPlatformExceptionEventProcessor();
+
+  @override
+  FutureOr<SentryEvent?> apply(SentryEvent event, {hint}) async {
+    final plaformException = event.throwable;
+    if (!(plaformException is PlatformException)) {
+      return event;
+    }
+
+    final nativeStackTrace = plaformException.stacktrace;
+    if (nativeStackTrace == null) {
+      return event;
+    }
+
+    // PackageInfo has an internal cache, so no need to do it ourselves.
+    final packageInfo = await PackageInfo.fromPlatform();
+    return _processPlatformException(
+      event,
+      plaformException,
+      nativeStackTrace,
+      packageInfo.packageName,
+    );
+  }
+
+  SentryEvent _processPlatformException(
+    SentryEvent event,
+    PlatformException exception,
+    String nativeStackTrace,
+    String packageName,
+  ) {
+    final e =
+        _JvmExceptionFactory(packageName).fromJvmStackTrace(nativeStackTrace);
+
+    return event.copyWith(
+      exceptions: [
+        ...?event.exceptions,
+        ...e,
+      ],
+    );
```

Comment:
What if we create a `SentryPlatformException` instead?
The problem here is that the `PlatformException`, the `value` field becomes a huge string, which is pretty much the same as the `nativeStackTrace` and this makes it very ugly.
Since we are converting it, we don't need to report that anymore.

Reply:
Yeah, I had the same problem with the Dio integration.

I would like to have a functionality where I have a map from an exception type to a function which returns a SentryEvent. Kinda like an event processor but instead of modifying events, it creates the event. 

That would have made things so much easier.

I'll make it look prettier 👍

Reply:
Done